### ISSUE TITLE
A word of advice

### ISSUE THREAD

**Issue by omrysendik** [sniklaus/pytorch-pwc#1]
Hi @sniklaus,

What do you propose as the best way to change your correlation kernel to correlate along the X and Y axis (rather than the C axis which AFAIU is what is currently done).

Thanks!

**Comment by sniklaus** [comment]
Thank you for your feedback. The implementation of the correlation kernel is equivalent to the one in the official PWC-Net repository. In fact, they are both modeled after the implementation in the FlowNet2 repository. I have added a comparison section in the readme to ease your concerns.

**Comment by omrysendik** [comment]
Yup... I apologize for the trivial question. I realized I can use an existing conv2d method from **torch.nn.functional**.
However, I would like to ask if you documented what's going on in this implementation? I might as well learn from it, after wasting a few precious hours before finding the existing method :-)
Any other decent documentation (CuPy documentation isn't decent...) would also be good.

**Comment by sniklaus** [comment]
Deqing gave a great explanation of the fundamental principles of cost volumes here: https://github.com/deqings/PWC-Net/issues/1

As for CuPy, you can find a more basic example here: https://github.com/sniklaus/pytorch-extension

**Comment by omrysendik** [comment]
Yeah, I was going through your Hadamard example :-)
I was looking for something a bit more documented but I'll probably go through it once again.

Thanks for the help!!!
